projects/storefrontlib/src/cms-structure/page/component/events/component.event.ts

```diff
@@ -0,0 +1,42 @@
+import { CxEvent } from '@spartacus/core';
+
+/**
+ * Lifecycle events for the creation and removal of CMS components.
+ *
+ * Triggers when the component is added or removed from the DOM, using
+ * sub event `ComponentCreateEvent` and `ComponentDestroyEvent`.
+ */
+export abstract class ComponentEvent extends CxEvent {
```

Comment:
Since this is only for _CMS_ components, does it make sense to name it _CmsComponentEvent_?

Reply:
A few of the concepts we build around (cms) component creation are widening up, which is why I thought about not naming it _CmsComponentEvent_. That being said, it would be quit useful to have that information, to distinguish components. I.e. if we'd make our SmartEdit integration based on these events, then we'd need to know what type of dynamic component we have. 

@dunqan thoughts?